test/sbvr/migrations.coffee

```diff
@@ -0,0 +1,111 @@
+fs = require('fs')
+typeVocab = fs.readFileSync(require.resolve('@resin/sbvr-types/Type.sbvr'))
+test = require('./test')(typeVocab)
+
+describe 'migrations', ->
+	test.migration '''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''',
+		'''
+			Term:      person
+		''', [ '''
+			ALTER TABLE "person"
+				DROP COLUMN IF EXISTS "name";\n\t
+		''' ]
+
+	test.migration '''
+			Term:      person
+		''', '''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''', [ '''
+		ALTER TABLE "person"
+			ADD COLUMN IF NOT EXISTS "name" VARCHAR(255) NOT NULL;\n\t
+		''' ]
+
+	test.migration '''
+			Term:      name
+				Synonym: full name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''', '''
+			Term:      full  name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has full name
+				Necessity: each person has exactly one full name
+		''', [ '''
+		ALTER TABLE "person"
+			RENAME COLUMN "name" TO "full name";\n\t
+		''' ]
+
+	test.migration '''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      person
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''',
+		'''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      years of experience
+				Concept Type: Integer (Type)
+			Term:      person
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+			Fact Type: person has years of experience
+				Necessity: each person has exactly one years of experience
+		''', [ '''
+		ALTER TABLE "person"
+			ADD COLUMN IF NOT EXISTS "years of experience" INTEGER NOT NULL;\n\t
+		''' ]
+
+	test.migration '''
+		Term: actor
+		''',
+		'''
+			Term: actor
+			Term: user
+				Concept Type: actor
+		''', [ '''
+		CREATE TABLE IF NOT EXISTS "user" (
+			"created at" TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+		,	"id" SERIAL NOT NULL PRIMARY KEY
+		,	"actor" INTEGER NOT NULL
+		,	FOREIGN KEY ("actor") REFERENCES "actor" ("id")
+		);
+		''' ]
+
+	test.migration '''
+		Term: pilot
+		Term: plane
+
+		Fact type:  pilot can fly plane
+			Synonymous Form: plane can be flown by pilot
+		''',
+		'''
+		Term: pilot
+		Term: plane
+
+		Fact type:  plane can be flown by pilot
+		''', [ '''
+			ALTER TABLE "pilot-can fly-plane"
+				RENAME TO "plane-can be flown by-pilot";
+			ALTER TABLE "plane-can be flown by-pilot"
+				RENAME COLUMN "pilot" TO "can be flown by-pilot";
+				RENAME COLUMN "can fly-plane" TO "plane";\n\t
```

Comment:
This still has the issue from https://github.com/resin-io-modules/abstract-sql-compiler/pull/13#discussion_r146668460
> Oh, also the change was referring to just the line I commented on, for reference I would expect the full block to be:
> ```coffee
> [ '''
> 	ALTER TABLE "pilot-can fly-plane"
> 	RENAME TO "plane-can be flown by-pilot";
> 	ALTER TABLE "plane-can be flown by-pilot"
> 	RENAME COLUMN "pilot" TO "can be flown by-pilot";
> 	ALTER TABLE "plane-can be flown by-pilot"
> 	RENAME COLUMN "can fly-plane" TO "plane";
> ''' ]
> ```
> or
> ```coffee
> [ '''
> 	ALTER TABLE "pilot-can fly-plane"
> 	RENAME TO "plane-can be flown by-pilot";
> ''', '''
> 	ALTER TABLE "plane-can be flown by-pilot"
> 	RENAME COLUMN "pilot" TO "can be flown by-pilot";
> ''', '''
> 	ALTER TABLE "plane-can be flown by-pilot"
> 	RENAME COLUMN "can fly-plane" TO "plane";
> ''' ]
> ```

Reply:
Ohh I kept misreading, I thought that renames on columns where in the list of actions. I will change that everywhere so that each rename has its own alter table